### PR TITLE
MM-52240: Stop/Start DB Cluster after load test

### PR DESCRIPTION
**Pull request by agnivade**
We add the ability to stop/start the DB cluster after a load-test is complete.

There is a dedicated command that the users can run any time. Additionally, there is also an automated way to stop the DB after a load test finishes, which is exposed via a flag.

Using the flag makes the load test runs synchronously, so it is advised to be used only by advanced users running the command preferably inside an EC2 instance in a screen session.

https://mattermost.atlassian.net/browse/MM-52240

### CODE REVIEW

File: cmd/ltctl/loadtest.go [hunk 1]
```diff
@@ -68,7 +73,75 @@ func RunLoadTestStartCmdF(cmd *cobra.Command, args []string) error {
...
+				return fmt.Errorf("failed to get DB status: %w", err)
+			}
+			if status == dbAvailable {
+				break
+			}
+			fmt.Println("Sleeping... ")
+			time.Sleep(30 * time.Second)
+		}
+	} else if status != dbAvailable {
+		fmt.Println("The database isn't available at the moment. It might be either stopping or starting. Please wait until it has fully stopped or started, and then try again.")
```
Comment (agarciamontoro) [inline cmd/ltctl/loadtest.go:104]: Can we just print the current status here? Something like
```suggestion
		fmt.Printf("The database isn't available at the moment. Its status is %q. Please wait until it has finished, and then try again.\n", status)
```

File: cmd/ltctl/loadtest.go [hunk 2]
```diff
@@ -68,7 +73,75 @@ func RunLoadTestStartCmdF(cmd *cobra.Command, args []string) error {
...
+		return nil
+	}
+
+	isSync, err := cmd.Flags().GetBool("sync")
+	if err != nil {
+		return fmt.Errorf("unable to check flag: %w", err)
```
Comment (agarciamontoro) [inline cmd/ltctl/loadtest.go:110]: Nit:
```suggestion
		return fmt.Errorf("unable to check -sync flag: %w", err)
```

File: cmd/ltctl/loadtest.go [hunk 3]
```diff
@@ -68,7 +73,75 @@ func RunLoadTestStartCmdF(cmd *cobra.Command, args []string) error {
...
+		if coordStatus.State == coordinator.Done {
+			fmt.Println("unbounded load-test has completed")
+			break
+		}
+
+		fmt.Println("Sleeping ...")
+		// Sleeping for 5 minutes gives 12 lines an hour.
+		// For an avg unbounded test of 4-5 hours, it gives around 50 lines,
+		// which should be acceptable.
+		time.Sleep(5 * time.Minute)
```
Comment (agarciamontoro) [inline cmd/ltctl/loadtest.go:140]: I agree it's acceptable :+1:

File: cmd/ltctl/loadtest.go [hunk 4]
```diff
@@ -68,7 +73,75 @@ func RunLoadTestStartCmdF(cmd *cobra.Command, args []string) error {
...
+	}
+
+	// Now we stop the DB.
+	return t.StopDB()
```
Comment (agarciamontoro) [inline cmd/ltctl/loadtest.go:144]: :heart:

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Changes look great! I've left some comments below, but mostly questions and nits. Thank you for this!

A more general question: do we always create a DB cluster, even if we have a single DB instance? The changes here assume there's always a cluster deployed.

File: deployment/terraform/db_operations.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+		"rds",
+		"stop-db-cluster",
+		"--db-cluster-identifier=" + output.DBCluster.ClusterIdentifier,
+		"--region=" + t.config.AWSRegion,
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	if err := t.runAWSCommand(ctx, args, nil); err != nil {
+		return err
+	}
```
[the remark this answers is not in the document]
Comment (agnivade) [reply deployment/terraform/db_operations.go:37]: Yes, the command is async. It does not wait for it to be fully stopped/started. So 30 secs is enough.

File: cmd/ltctl/loadtest.go [hunk 6]
```diff
@@ -68,7 +73,75 @@ func RunLoadTestStartCmdF(cmd *cobra.Command, args []string) error {
...
+	// Now we keep checking the status of the coordinator until it's done.
+	var coordStatus coordinator.Status
+	for {
+		coordStatus, err = t.GetCoordinatorStatus()
+		if err != nil {
+			return err
+		}
+
+		if coordStatus.State == coordinator.Done {
+			fmt.Println("unbounded load-test has completed")
```
[the remark this answers is not in the document]
Comment (agnivade) [reply cmd/ltctl/loadtest.go:132]: Yep for sure. Once it reaches the limit, after a while, it will automatically stop.

**Comment by agnivade** [comment]
> A more general question: do we always create a DB cluster, even if we have a single DB instance? The changes here assume there's always a cluster deployed.

Yep, we always create a cluster.

**Review by agarciamontoro — APPROVED** [review]
Thank you! :rocket:

File: cmd/ltctl/loadtest.go [hunk 7]
```diff
@@ -68,7 +73,75 @@ func RunLoadTestStartCmdF(cmd *cobra.Command, args []string) error {
...
+		for {
+			status, err := t.DBStatus()
+			if err != nil {
+				return fmt.Errorf("failed to get DB status: %w", err)
+			}
+			if status == dbAvailable {
+				break
+			}
+			fmt.Println("Sleeping... ")
+			time.Sleep(30 * time.Second)
```
Comment (isacikgoz) [inline cmd/ltctl/loadtest.go:101]: Isn't 30 second too much for the ping interval?

**Review by isacikgoz — APPROVED** [review]
LGTM, just a question

File: cmd/ltctl/loadtest.go (hunk 7 above)
Comment (agnivade) [reply cmd/ltctl/loadtest.go:101]: Not really. These take a long time to stop or start. In the order of ~5 minutes.

**Pull request merged**